fpsd/features.py

```diff
@@ -0,0 +1,765 @@
+#!/usr/bin/env python3.5
+import pandas as pd
+import os
+from sqlalchemy import create_engine
+from tqdm import tqdm
+import subprocess
+
+
+def compute_bursts(df):
+    """Compute list of burst lengths for a given trace
+    """
+    bursts = []
+    current_burst_length = 0
+
+    for row in df.itertuples():
+        if row.ingoing == False:
+            current_burst_length += 1
+        elif row.ingoing == True and current_burst_length > 0:
+            bursts.append(current_burst_length)
+            # Reset burst length
+            current_burst_length = 0
+
+    # If we were in a burst at the end of the trace, let's save it
+    if current_burst_length > 0:
+        bursts.append(current_burst_length)
+
+    # For some features we need to know the ordering of bursts,
+    # so let's save this information as well
+    ranks = list(range(1, len(bursts) + 1))
```

Comment:
Still just picking up SQL. I know now that databases don't preserve order of rows, so this is necessary.